jsettlers.logic/src/main/java/jsettlers/ai/highlevel/WhatToDoAi.java

```diff
@@ -266,27 +290,53 @@ private boolean isWoodJam() {
 				playerId) > aiStatistics.getNumberOfBuildingTypeForPlayer(LUMBERJACK, playerId) * 2;
 	}
 
-	private void destroyHinterlandMilitaryBuildings() {
-		for (ShortPoint2D militaryBuildingPositions : aiStatistics.getHinterlandMilitaryBuildingPositionsOfPlayer(playerId)) {
-			taskScheduler.scheduleTask(new DestroyBuildingGuiTask(playerId, militaryBuildingPositions));
-		}
-	}
-
 	private void buildBuildings() {
 		if (aiStatistics.getNumberOfNotFinishedBuildingsForPlayer(playerId) < economyMinister.getNumberOfParallelConstructionSites()) {
 			if (economyMinister.automaticLivingHousesEnabled() && buildLivingHouse())
 				return;
-			if (economyMinister.automaticTowersEnabled() && aiStatistics.getNumberOfNotOccupiedMilitaryBuildings(playerId) == 0 && buildTower())
-				return;
 			if (isEndGame) {
 				return;
 			}
+			if (buildTower()) {
+				return;
+			}
 			if (buildStock())
 				return;
 			buildEconomy();
 		}
 	}
 
+	private boolean buildTower() {
+		for (ShortPoint2D towerPosition : aiStatistics.getBuildingPositionsOfTypeForPlayer(TOWER, playerId)) {
+			Building tower = aiStatistics.getBuildingAt(towerPosition);
+			if (!tower.isConstructionFinished() || !tower.isOccupied()) {
+				return false;
+			}
+		}
+
+		List<ShortPoint2D> threatenedBorder = aiStatistics.threatenedBorderOf(playerId);
+		if (threatenedBorder.size() == 0) {
+			return false;
+		}
+
+		ShortPoint2D position = bestConstructionPositionFinderFactory
+				.getBorderDefenceConstructionPosition(threatenedBorder)
+				.findBestConstructionPosition(aiStatistics, mainGrid.getConstructionMarksGrid(), playerId);
+		if (position != null) {
+			taskScheduler.scheduleTask(new ConstructBuildingTask(EGuiAction.BUILD, playerId, position, TOWER));
```

Comment:
@codingberlin: What do you think about increasing the tower's build priority. The tower is build because of a border that needs to be defended. Therefore I think it could help to speed up this build by increaing it's priority.